### ISSUE TITLE
Memory consumption of HTTP responses in Vega

### ISSUE THREAD

**Issue by dma** [subgraph/Vega#4]
Original ticket:

> The scanner download all the file and store them in memory, this can be resource >exhausting.
> 
> By example by scanning ​http://recon.cx the scanner download hundred of meg of pdf >files, and end up exausting alot of bandwidth and system ressource.
> 
> Eventually the scanner ended up freezing...

Follow-up and deferral (brl):

The conclusion in the ticket description is incorrect: Vega does _not_ store all responses in memory. Some other memory performance problems with the scanner have been addressed which were causing Vega to excessively use memory as a scan progressed.

There is another issue which needs to be solved, which is that every individual response _is_ stored in memory until the entire response has been received. This means that Vega requires at least as much memory as the largest file it downloads, and potentially even more if large files are downloaded in parallel.

The eventual solution to this will probably be to set a limit under which responses will be stored entirely in memory. If this limit is exceeded, a file should be created and if the response is stored in the request log, this file should be available as a file entity linked to the response object in the database.
